app/src/main/java/br/com/bootcamp/magicgamecs/core/widgets/CardImageView.kt

```diff
@@ -0,0 +1,169 @@
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+import kotlin.math.roundToInt
+
+
+class CardImageView
+@JvmOverloads
+constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    FrameLayout(context, attrs, defStyleAttr) {
+
+    init {
+        inflate(context, R.layout.widget_cardimage, this)
+        attrs?.init()
+    }
+
+    private fun AttributeSet.init() {
+        val typedArray = context.obtainStyledAttributes(this, R.styleable.CardImageView, 0, 0)
+        val textSize =
+            typedArray.getDimensionPixelSize(R.styleable.CardImageView_civ_captionTextSize, 0)
+        text_card_placeholder.setTextSize(TypedValue.COMPLEX_UNIT_PX, textSize.toFloat())
+        typedArray.recycle()
+    }
+
+    var card: Card? = null
+        set(value) {
+            field = value
+            updateCard(value)
+        }
+
+    private fun updateCard(card: Card?) {
+        if (card == null) return
+        contentDescription = card.name
+        Glide.with(context)
+            .load(card.imageUrl)
+            .placeholder(R.drawable.placeholder_card)
+            .error(R.drawable.placeholder_card)
+            .fallback(R.drawable.placeholder_card)
+            .transform(RoundedCorners(0.05f))
+            .addListener(
+                CardImageRequestListener(
+                    card,
+                    shimmer_card,
+                    text_card_placeholder
+                )
+            )
+            .into(imageView_card_item)
+    }
+
+    class CardImageRequestListener(
+        private val card: Card,
+        private val shimmer: ShimmerFrameLayout,
+        private val textPlaceholder: TextView
+    ) : RequestListener<Drawable> {
+
+        init {
+            textPlaceholder.text = ""
+            startShimmer()
+        }
+
+        override fun onLoadFailed(
+            e: GlideException?,
+            model: Any?,
+            target: Target<Drawable>?,
+            isFirstResource: Boolean
+        ): Boolean {
+            textPlaceholder.text = card.name
+            stopShimmer()
+            return false
+        }
+
+        override fun onResourceReady(
+            resource: Drawable?,
+            model: Any?,
+            target: Target<Drawable>?,
+            dataSource: DataSource?,
+            isFirstResource: Boolean
+        ): Boolean {
+            stopShimmer()
+            return false
+        }
+
+        private fun startShimmer() {
+            shimmer.setShimmer(
+                Shimmer.AlphaHighlightBuilder()
+                    .setAutoStart(true)
+                    .setDuration(1300)
+                    .setRepeatMode(ValueAnimator.RESTART)
+                    .build()
+            )
+        }
+
+        private fun stopShimmer() {
+            shimmer.stopShimmer()
+            shimmer.setShimmer(null)
+            shimmer.clearAnimation()
+        }
+
+    }
+
+    class RoundedCorners(
```

Comment:
Acredito que esta classe não está diretamente relacionada às demais implementações deste arquivo. Minha sugestão é movê-lo para um outro arquivo.

Reply:
isso dentro do próprio package `widgets` mesmo?

Reply:
Alterado

Reply:
Depende de como vc trata o package de widget. Se ele tudo que tiver relação com os widget sim, se for apenas as customviews e views customizadas acredito que não.